### ISSUE TITLE
Installation fails due to conflicting openpyxl version

### ISSUE THREAD

**Issue by NeolithEra** [wind39/spartacus#19]
Hi, users are unable to run **_Spartacus_** due to dependency conflict with _**openpyxl**_ package. As shown in the following full dependency graph of **_Spartacus_**, **_Spartacus_** requires _**openpyxl***_，while **_pyexcel-xlsx_** requires **_openpyxl <2.6.0,>=2.5.0_**.

According to pip’s _“first found wins”_ installation strategy, _**openpyxl 3.0.2**_ is the actually installed version. However, _**openpyxl 3.0.2**_ does not satisfy **_<2.6.0,>=2.5.0_**.

### Dependency tree-----------
```
Spartacus - 3.36
| +- bs4(install version:0.0.1 version range:*)
| | +- beautifulsoup4(install version:4.8.1 version range:*)
| | | +- soupsieve(install version:1.9.5 version range:>=1.2)
| +- click(install version:7.0 version range:*)
| +- cryptography(install version:2.8 version range:*)
| +- lxml(install version:4.4.2 version range:*)
| +- openpyxl(install version:3.0.2 version range:*)
| +- pillow(install version:6.2.1 version range:*)
| | +- black(install version:19.10b0 version range:*)
| | | +- appdirs(install version:1.4.3 version range:*)
| | | +- attrs(install version:19.3.0 version range:>=18.1.0)
| | | +- click(install version:7.0 version range:>=6.5)
| | | +- pathspec(install version:0.6.0 version range:<1,>=0.6)
| | | +- regex(install version:2019.12.20 version range:*)
| | | +- toml(install version:0.10.0 version range:>=0.9.4)
| | | +- typed-ast(install version:1.4.0 version range:>=1.4.0)
| | +- check-manifest(install version:0.40 version range:*)
| | | +- toml(install version:0.10.0 version range:*)
| | +- coverage(install version:5.0 version range:*)
| | +- coveralls(install version:1.9.2 version range:*)
| | +- jarn.viewdoc(install version:2.3 version range:*)
| | +- olefile(install version:0.46 version range:*)
| | +- pycodestyle(install version:2.5.0 version range:*)
| | +- pyflakes(install version:2.1.1 version range:*)
| | +- pyroma(install version:2.6 version range:*)
| | | +- docutils(install version:0.16rc1 version range:*)
| | | +- pygments(install version:2.5.2 version range:*)
| | | +- setuptools(install version:42.0.2 version range:*)
| | +- pytest(install version:5.3.2 version range:*)
| | +- pytest-cov(install version:2.8.1 version range:*)
| | | +- coverage(install version:5.0 version range:>=4.4)
| | | +- pytest(install version:5.3.2 version range:>=3.6)
| | +- sphinx-rtd-theme(install version:0.4.3 version range:*)
| | | +- sphinx(install version:2.3.0 version range:*)
| +- pyaes(install version:1.6.1 version range:*)
| +- pyexcel(install version:0.5.15 version range:*)
| | +- lml(install version:0.0.9 version range:>=0.0.4)
| | +- pyexcel-io(install version:0.5.20 version range:>=0.5.19)
| | | +- lml(install version:0.0.9 version range:>=0.0.4)
| +- pyexcel-xls(install version:0.5.8 version range:*)
| | +- pyexcel-io(install version:0.5.20 version range:>=0.5.3)
| | | +- lml(install version:0.0.9 version range:>=0.0.4)
| | +- xlrd(install version:1.2.0 version range:*)
| | +- xlwt(install version:1.3.0 version range:*)
| | | +- coveralls(install version:1.9.2 version range:*)
| | | +- nose(install version:1.3.7 version range:*)
| | | +- nose-cov(install version:1.6 version range:*)
| | | +- nose-fixes(install version:1.3 version range:*)
| | | | +- nose(install version:1.3.7 version range:*)
| | | | +- setuptools(install version:42.0.2 version range:*)
| | | +- panci(install version:0.0.2 version range:*)
| | | | +- pyyaml(install version:5.2 version range:*)
| | | +- pkginfo(install version:1.5.0.1 version range:*)
| | | +- setuptools-git(install version:1.2 version range:*)
| | | +- six(install version:1.13.0 version range:*)
| | | +- sphinx(install version:2.3.0 version range:*)
| | | +- tox(install version:3.14.2 version range:*)
| | | +- twine(install version:3.1.1 version range:*)
| | | +- wheel(install version:0.33.6 version range:*)
| +- pyexcel-xlsx(install version:0.5.7 version range:*)
| | +- openpyxl(install version:2.5.14 version range:>=2.5.0,<2.6.0)
| | | +- et-xmlfile(install version:1.0.1 version range:*)
| | | | +- python(install version:2.5 version range:>=2.6.0)
| | | +- jdcal(install version:1.4.1 version range:*)
| | +- pyexcel-io(install version:0.5.20 version range:>=0.5.3)
| | | +- lml(install version:0.0.9 version range:>=0.0.4)
| +- pyscrypt(install version:1.6.2 version range:*)
| +- sqlparse(install version:0.3.0 version range:*)
| +- tabulate(install version:0.8.6 version range:*)
```

Thanks for your help.
Best,
Neolith

**Comment by NeolithEra** [comment]
### Suggested Solution

1.	Fix your direct dependency to be **_openpyxl <2.6.0_**. I have checked this revision will not affect your downstream projects now.
2.	Remove your direct dependency **_openpyxl_**, and use **_openpyxl_** transitively introduced by **_pyexcel-xlsx_**.

3.	Ask your upstream project **_pyexcel-xlsx_** to loose the version range of **_openpyxl_**.

@wind39 Which solution do you prefer, 1, 2 or 3?
Please let me know your choice. I can submit a PR to solve this issue.

**Comment by wind39** [comment]
Hi @NeolithEra ,

Thank you very much for reporting.

@barthisrael , do you happen to know why `pyexcel-xlsx` requires `openpyxl <2.6.0,>=2.5.0` ?

Thanks and regards,

William

**Comment by wind39** [comment]
Hi,

It's been recently discussed here: https://github.com/pyexcel/pyexcel-xlsx/issues/34

The original issue is here: https://github.com/pyexcel/pyexcel-io/issues/66

This is why they pinned `openpyxl < 2.6.0`. Not sure if newer versions would work, though.

Regards,

William

**Comment by wind39** [comment]
Hi @NeolithEra ,

`Spartacus.Utils.DataFileReader` requires `pyexcel` to be able to read `.xls` files.

I just released Spartacus 3.38, where support to reading and writing to both `xlsx` and `xls` file formats is optional.

So now following the same concept of installing support to a specific database management system, for example:

```bash
pip install Spartacus[postgresql]
```

You can also install support to `xls` or `xlsx` if required, i.e.:

```
pip install Spartacus[xls]
```

Installing support to `xlsx` installs recent `openpyxl`. Installing support to `xls` installs `pyexcel` and whatever version of `openpyxl` that `pyexcel` requires (and also enables support to `xlsx`).

Please let me know if this fixes the issue you are seeing.

Best regards,

William

**Comment by wind39** [comment]
Hi @NeolithEra ,

I'm happy to report that this issue was fixed upstream in `pyexcel`:

https://github.com/pyexcel/pyexcel-xlsx/issues/34

So that would be option 3 in your list of suggestions.

Thank you very much, now I'm closing this issue.

Best regards,

William
